x-pack/plugins/apm/server/routes/metrics/serverless/helper.ts

```diff
@@ -33,3 +37,65 @@ export function calcMemoryUsed({
 
   return memoryTotal - memoryFree;
 }
+
+const GB = 1024 ** 3;
+/**
+ * To calculate the compute usage we need to multiply the "system.memory.total" by "faas.billed_duration".
+ * But the result of this calculation is in Bytes-milliseconds, as the "system.memory.total" is stored in bytes and the "faas.billed_duration" is stored in milliseconds.
+ * But to calculate the overall cost AWS uses GB-second, so we need to convert the result to this unit.
+ */
+export function calcComputeUsageGBSeconds({
+  billedDuration,
+  totalMemory,
+}: {
+  billedDuration?: number | null;
+  totalMemory?: number | null;
+}) {
+  if (!isFiniteNumber(billedDuration) || !isFiniteNumber(totalMemory)) {
+    return 0;
+  }
+
+  const totalMemoryGB = totalMemory / GB;
+  const billedDurationSec = billedDuration / 1000;
+  return totalMemoryGB * billedDurationSec;
+}
+
+export function calcEstimatedCost({
+  awsLambdaPriceFactor,
+  architecture,
+  transactionThroughput,
+  billedDuration,
+  totalMemory,
+  awsLambdaRequestCostPerMillion,
+}: {
+  awsLambdaPriceFactor?: AWSLambdaPriceFactor;
+  architecture?: AwsLambdaArchitecture;
+  transactionThroughput: number;
+  billedDuration?: number | null;
+  totalMemory?: number | null;
+  awsLambdaRequestCostPerMillion?: number;
+}) {
+  try {
+    if (
+      !awsLambdaPriceFactor ||
+      !architecture ||
+      !awsLambdaRequestCostPerMillion ||
+      !awsLambdaPriceFactor?.[architecture]
```

Comment:
I think this should not return 0, and also we should be careful here, because `!0` is also `true`. We should either explicitly check for `null` or use `isFiniteNumber`.